frontend/src/editor/speaker_dropdown.tsx

```diff
@@ -1,83 +1,78 @@
-import { useMemo, useState } from 'react';
-import { useSlate, ReactEditor } from 'slate-react';
+import { memo, useState, useCallback } from 'react';
+import { Editor } from 'slate';
+import { useSlateStatic, ReactEditor } from 'slate-react';
 import * as Automerge from '@automerge/automerge';
-import { AutomergeEditor } from 'slate-automerge-doc';
 
 import { Document, Paragraph } from './types';
 import { PrimaryButton, SecondaryButton } from '../components/button';
 import { IoIosCreate, IoIosList, IoIosTrash } from 'react-icons/io';
 import { Dropdown, DropdownItem, DropdownSection } from '../components/dropdown';
 import { Input, Select } from '../components/form';
+import { useSpeakerName, useSpeakerNames } from '../utils/document';
 import { showModal, Modal, ModalProps } from '../components/modal';
 
-export function getSpeakerName(element: Paragraph, speaker_names: Record<string, string>): string {
-  if (!element.speaker) {
-    return `Unknown Speaker`;
-  } else if (element.speaker in speaker_names) {
-    return speaker_names[element.speaker];
-  } else {
-    return `Unnamed Speaker ${element.speaker}`;
-  }
-}
-
 function SelectSpeakerModal({
-  doc,
+  editor,
   onSpeakerSelected,
   onNewSpeaker,
   onClose,
   selected,
   ...props
 }: {
-  doc: Automerge.Doc<Document>;
+  editor: Editor;
```

Comment:
we can use useSlate / useSlateStatic here

Reply:
we cannot, the modals are not below slate in the react tree